src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointComplications.java

```diff
@@ -0,0 +1,503 @@
+         * 13: GGGCAGCTGTGGA
+         * 39: TGGTGCAAATGCCATTTATGCTCCTCTCCACCCATATCC
+         * And the arithmetic works this way:
+         * 31 + 96 = 127
+         * 96 + 13 + 39 = 148
+         */
+        @VisibleForTesting
+        TandemRepeatComplication(final int distBetweenAlignRegionsOnRef, final int distBetweenAlignRegionsOnCtg) {
+            // the reference system with a shorter overlap (i.e. with less-negative distance between regions) has a higher repeat number
+            final boolean isExpansion = distBetweenAlignRegionsOnRef < distBetweenAlignRegionsOnCtg;
+            final int overlapOnLowerCNSequence, overlapOnHigherCNSequence;
+            if (isExpansion) {
+                overlapOnLowerCNSequence = Math.abs(distBetweenAlignRegionsOnRef);
+                overlapOnHigherCNSequence = Math.abs(distBetweenAlignRegionsOnCtg);
+            } else {     // d1 is lower absolute value -> reference has higher copy number of the duplication, i.e. Deletion
+                overlapOnLowerCNSequence = Math.abs(distBetweenAlignRegionsOnCtg);
+                overlapOnHigherCNSequence = Math.abs(distBetweenAlignRegionsOnRef);
+            }
+
+            int higherCnEst=0, lowerCnEst=0, unitLen=0, pseudoHomLen=0;
+            double err = Double.MAX_VALUE;
+            for(int cn2 = 1; cn2< MAX_LOWER_CN; ++cn2) {
+                for(int cn1=cn2+1; cn1<=2*cn2; ++cn1) {
+                    final int dupLenUpperBound = cn1==2*cn2 ? overlapOnLowerCNSequence : overlapOnHigherCNSequence;
+                    for (int l=2; l<=dupLenUpperBound; ++l) {
+                        for (int lambda=0; lambda<l; ++lambda) {
+                            final int d1 = (2*cn2 - cn1)*l + lambda;
+                            final int d2 = cn2*l + lambda;
+                            final double newErr = Math.abs(overlapOnHigherCNSequence-d1) + Math.abs(overlapOnLowerCNSequence-d2);
+                            if (newErr < err) {
+                                err = newErr;
+                                higherCnEst = cn1; lowerCnEst = cn2;
+                                unitLen= l; pseudoHomLen = lambda;
+                            }
+                            if (err<1){
+                                lowerRepeatNumberEstimate = lowerCnEst;
+                                higherRepeatNumberEstimate = higherCnEst;
+                                repeatedSeqLen = unitLen;
+                                pseudoHomologyLen = pseudoHomLen;
+                                return;
+                            }
+                        }
+                    }
+                }
+            }
+
+            lowerRepeatNumberEstimate = lowerCnEst;
+            higherRepeatNumberEstimate = higherCnEst;
+            repeatedSeqLen = unitLen;
+            pseudoHomologyLen = pseudoHomLen;
+        }
+    }
+
+
+    BreakpointComplications(final String homologyForwardStrandRep, final String insertedSequenceForwardStrandRep,
```

Comment:
OK. I might put in a comment that says the constructor is only for test purposes, then.